Update green mode docs and asyncio example

ajoubertza opened the pull request:
These changes are based on comments in issue #333.

The fact that "asyncio is preferred" was agreed to at the Tango Kernel Developers meeting on [27/02/2020](https://github.com/tango-controls/tango-kernel-followup/blob/master/2020/2020-02-27/Minutes.md#pytango-news).  gevent and concurrent.futures are still supported, but we favour the asyncio approach going forward as it is the Python standard.

ajoubertza committed:
Update green mode docs and asyncio example

These are based in comments in issue 333.

vxgmichel commented on examples/asyncio_green_mode/asyncio_device_example.py:
```diff
@@ -14,6 +14,12 @@ class AsyncioDevice(Device):
+        self.set_state(DevState.CLOSE)
+
+    @command
+    async def background_task_command(self):
```
:+1: for keeping both!

vxgmichel commented on doc/green_modes/green_modes_server.rst:
```diff
@@ -26,11 +22,10 @@ as it is a lot harder to debug. You should use this green mode with care.
+green mode there is no Tango monitor lock!
 
-Greenlets (a task in a background, but handled within the event loop) can be
+Greenlets (a task in the background, but handled within the event loop) can be
```
Two suggestions:

     Another thing to keep in mind is that when using :obj:`~tango.GreenMode.Gevent` green mode 
     the Tango monitor lock is disabled, so the client requests can be processed concurrently.

and: 

    Greenlets can also be used to spawn tasks in the background.

ajoubertza replied:
Thanks.  Updated in 9e0a2d9.

ajoubertza committed:
Update gevent server doc as per PR comments

ajoubertza committed:
Add note on preferred green mode (asyncio) to docs

This was agreed to at the Tango Kernel Developers meeting on 27/02/2020.
gevent and concurrent.futures are still supported, but we favour the
asyncio approach going forward as it is the Python standard.
